projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -260,11 +260,39 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
      * @hidden
      */
     protected groupingDiffer;
+    protected _rowSelectionMode: GridSelectionMode;
     private _data;
     private _hideGroupedColumns = false;
     private _dropAreaMessage = null;
     private _showGroupArea = true;
 
+    /**
+     * Gets/Sets row selection mode
+     *
+     * @remarks
+     * By default the row selection mode is none
+     * @param selectionMode: FlatGridSelectionMode
+     */
+    @Input()
+    get rowSelection() {
+        return this._rowSelectionMode;
+    }
+
+    set rowSelection(selectionMode: GridSelectionMode) {
+        this._rowSelectionMode = selectionMode;
+        if (!this._init) {
+            this.selectionService.clearAllSelectedRows();
+            this.notifyChanges(true);
+        }
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get isMultiRowSelectionEnabled(): boolean {
```

Comment:
You don't need this. Leave the base implementation take care of it. It will return true on `GridSelectionMode.multiple` and on `GridSelectionMode.multipleCascade`

Reply:
Fair enough, there's very little reason to override this check indeed